scanpointgenerator/excluders/roiexcluder.py

```diff
@@ -0,0 +1,67 @@
+from scanpointgenerator.core import Excluder, ROI
+from scanpointgenerator.compat import np
+
+
+@Excluder.register_subclass("scanpointgenerator:excluder/ROIExcluder:1.0")
+class ROIExcluder(Excluder):
+
+    """A class to exclude points outside of regions of interest."""
+
+    def __init__(self, rois, axes):
+        """
+        Args:
+            rois(list(ROI)): List of regions of interest
+            axes(list(str)): Names of axes to exclude points from
+
+        """
+        super(ROIExcluder, self).__init__(axes)
+
+        self.rois = rois
+
+    def create_mask(self, x_points, y_points):
+        """Create a boolean array specifying the points to exclude.
+
+        The resulting mask is created from the union of all ROIs.
+
+        Args:
+            x_points(numpy.array(float)): Array of points for x-axis
+            y_points(numpy.array(float)): Array of points for y-axis
+
+        Returns:
+            np.array: Boolean array of points to exclude
+
+        """
+        if len(x_points) != len(y_points):
+            raise ValueError("Points lengths must be equal")
+
+        mask = np.zeros_like(x_points, dtype=bool)
```

Comment:
`dtype=np.int8`

The in-place bitwise operations don't work for the `bool` type in the Jython numpy emulation.